init.bin - add optional support for zeal-coreutils

zoul0813 opened the pull request:
* add menuconfig option to enable zeal-coreutils
* omit zeal-coreutils commands from init.bin shell

zoul0813 committed:
* add menuconfig option to enable zeal-coreutils
* omit zeal-coreutils commands from init.bin shell

Zeal8bit commented on romdisk/Kconfig:
```diff
@@ -0,0 +1,12 @@
+                prompt "Enable zeal-coreutils"
+                default n
+                help
+                        Boolean to mark whether the target has an MMU/Memory Mapper. If this option is selected,
```
The description doesn't match the option

Zeal8bit commented on romdisk/parse.asm:
```diff
@@ -520,6 +520,19 @@ help_msg_end:
 
         SECTION DATA
 system_commands_begin:
+    IF CONFIG_ENABLE_COREUTILS > 0
```
```suggestion
    IF CONFIG_ENABLE_COREUTILS
```
`IF` condition in z88dk-z80asm implies a non-zero value

Zeal8bit commented on romdisk/Makefile:
```diff
@@ -24,13 +32,24 @@ LIBS=-L$(ZOS_PATH)/kernel_headers/z88dk-z80asm/lib
 .PHONY: all clean
 
 all: clean
+	@echo "SRCS: " $(SRCS)
```
Debug leftover? :smile:

zoul0813 committed:
* fix menuconfig description
* remove debug print and unnecessary condition check

zoul0813 committed:
* add less,xfer,dir back into init.bin
* update configs, fix Makefile

zoul0813 committed:
fix: zealemu.default had duplicate ENABLE_TFCARD

zoul0813 committed:
update: moved "Executable to launch" into new "Shell" menu
add: "ROMDISK_EXTRA_FILES" to KConfig
update: pack.py now supports ENV{EXTRA_ROMDISK_FILES}, CONFIG{EXTRA_ROMDISK_FILES} and args
update: pack.py now supports being passed a dir, it expands all files in the dir
remove: ${EXTRA_ROMDISK_FILES} env var from Makefile
update: Makefile now makes KCONFIG_CONFIG an abs path so pack.py can use it
update: default configs now account for romdisk menu, with standard setup

zoul0813 committed:
remove `dir` alias to `ls`, coreutils does not implement this alias however

Zeal8bit committed:
romdisk: make the romdisk address relative to the kernel, no need to provide physical addresses anymore

Zeal8bit committed:
build: trigger a rebuild when the configuration changes

zoul0813 committed:
add zos_link_libraries to support linking to library targets in the project

zoul0813 committed:
make the target depend on the library file

Zeal8bit committed:
cmake: fix typos
